spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -303,29 +370,38 @@ object SparkTableUtil {
     }
   }
 
-  private def buildManifest(conf: SerializableConfiguration,
-      sparkDataFiles: Seq[SparkDataFile],
-      partitionSpec: PartitionSpec,
-      basePath: String): Iterator[Manifest] = {
-    if (sparkDataFiles.isEmpty) {
-      Seq.empty.iterator
-    } else {
+  private def toSparkPartition(partition: CatalogTablePartition, table: CatalogTable): SparkPartition = {
+    val uri = partition.storage.locationUri.map(String.valueOf(_))
+    require(uri.nonEmpty, "Partition URI should be defined")
+
+    val format = partition.storage.serde.orElse(table.provider)
```

Comment:
We use `table.provider` if `partition.storage.serde` is not set (happens with the in-memory catalog).